RecoBTag/ChargeTagging/src/CandidateChargeBTagComputer.cc

```diff
@@ -0,0 +1,243 @@
+	
+	float sv_ch_weight = pow(trackPtr->pt(),svChargeExp_);
+	sv_ch_num += sv_ch_weight * trackPtr->charge();
+	sv_ch_den += sv_ch_weight;
+      }
+      
+      if ( sv_ch_den > 0 ) sv_ch = sv_ch_num / sv_ch_den;
+    }
+
+    // fill soft muon variables
+    if ( n_sm_info > 0 )
+    {
+      // find the muon with higher transverse momentum
+      int lep_idx = 0;
+      float max_pt = 0;
+      for (size_t i_lep=0; i_lep<n_sm_info; ++i_lep)
+      {
+	float lep_pt = sm_info.lepton(0)->pt();
+	if ( lep_pt > max_pt )
+	{
+	  max_pt = lep_pt;
+	  lep_idx = i_lep;
+	}
+      }
+      
+      mu_sip2d = sm_info.properties(lep_idx).sip2d;
+      //mu_sip3d = sm_info.properties(lep_idx).sip3d;
+      mu_delR  = sm_info.properties(lep_idx).deltaR;
+      mu_ptrel = sm_info.properties(lep_idx).ptRel;
+      mu_pfrac = sm_info.properties(lep_idx).ratio;
+      mu_char  = sm_info.properties(lep_idx).charge;
+    }
+      
+    // fill soft electron variables
+    if ( n_se_info > 0 )
+    {
+      // find the electron with higher transverse momentum
+      int lep_idx = 0;
+      float max_pt = 0;
+      for (size_t i_lep=0; i_lep<n_se_info; ++i_lep)
+      {
+	float lep_pt = se_info.lepton(0)->pt();
+	if ( lep_pt > max_pt )
+	{
+	  max_pt = lep_pt;
+	  lep_idx = i_lep;
+	}
+      }
+      
+      //el_sip2d = se_info.properties(lep_idx).sip2d;
+      //el_sip3d = se_info.properties(lep_idx).sip3d;
+      el_delR  = se_info.properties(lep_idx).deltaR;
+      el_ptrel = se_info.properties(lep_idx).ptRel;
+      el_pfrac = se_info.properties(lep_idx).ratio;
+      el_mva   = se_info.properties(lep_idx).elec_mva;
+      el_char  = se_info.properties(lep_idx).charge;
+    }
+      
+    std::map<std::string,float> inputs;
+    inputs["tr_ch"]    = tr_ch;
+    inputs["pt1_ch/j_pt"] = pt_ratio1_ch;
+    inputs["pt2_ch/j_pt"] = pt_ratio2_ch;
+    inputs["pt3_ch/j_pt"] = pt_ratio3_ch;
+    inputs["sv_ch"]    = sv_ch;
+    inputs["mu_sip2d"] = mu_sip2d;
+    //inputs["mu_sip3d"] = mu_sip3d;
+    inputs["mu_delR"]  = mu_delR;
+    inputs["mu_ptrel"] = mu_ptrel;
+    inputs["mu_pfrac"] = mu_pfrac;
+    inputs["mu_char"]  = mu_char;
+    //inputs["el_sip2d"] = el_sip2d;
+    //inputs["el_sip3d"] = el_sip3d;
+    inputs["el_delR"]  = el_delR;
+    inputs["el_ptrel"] = el_ptrel;
+    inputs["el_pfrac"] = el_pfrac;
+    inputs["el_mva"]   = el_mva;
+    inputs["el_char"]  = el_char;
+
+    // evaluate the MVA
+    value = mvaID->evaluate(inputs);
+  }
+
+  // return the final discriminator value
+  return value;
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module ------------
+void
+CandidateChargeBTagComputer::fillDescriptions(edm::ConfigurationDescriptions & descriptions) {
+
+  edm::ParameterSetDescription desc;
+  desc.add<bool>("useCondDB",false);
+  desc.add<std::string>("gbrForestLabel","");
+  desc.add<edm::FileInPath>("weightFile",edm::FileInPath());
+  desc.add<bool>("useGBRForest",true);
```

Comment:
In the last round of comments I implied to remove the option and just use GBRForest